src/algorithms/controllers/msort_arr_bup.js

```diff
@@ -215,168 +194,191 @@ export function run_msort() {
       }
     }, [A, B, size], length);
 
-    chunker.add('runlength', (vis, runlength) => {
-
-      displayRunlength(vis, runlength, size);
-
+    chunker.add('runlength', (vis, cur_rlength, cur_size) => {
+      displayRunlength(vis, cur_rlength, cur_size);
     }, [runlength, size]);
 
     while (runlength < size + 1) {
       let left = 0;
 
-      chunker.add('MainWhile', (vis, runlength, size) => {
-        let size_txt = "size = " + (size + 1);
-        console.log("size = " + size);
-        assignVarToA(vis, size_txt, size);
-        displayRunlength(vis, runlength, size);
-      }, [runlength, size]);
-
-      chunker.add('left', (vis, a, cur_left, cur_mid, cur_right) => {
-        /*for (let i = cur_left; i <= cur_right; i++) {
-          unhighlight(vis, i, true);
-        }*/
-        assignVarToA(vis, 'left', cur_left);
+      chunker.add('MainWhile', (vis, cur_size, cur_rlength) => {
+        let size_txt = "size = " + (cur_size + 1);
+        assignVarToA(vis, size_txt, cur_size);
 
-      }, [A, left]);
+        for (let i = 0; i < cur_rlength; i++) {
+          highlight(vis, i, true);
+        }
+        for (let j = cur_rlength; j < Math.min(cur_rlength * 2, cur_size + 1); j++) {
+          highlight(vis, j, false);
+        }
+
+      }, [size, runlength]);
+
+
+      chunker.add('left', (vis, cur_left) => {
+        assignVarToA(vis, 'left', cur_left);
+      }, [left]);
 
 
       while ((left + runlength) < size + 1) {
 
```

Comment:
This could be refactored to `merge(left, mid, right)` but is it necessary, I don't think it is